test/11-api-binder.spec.coffee

```diff
@@ -135,3 +135,25 @@ describe 'APIBinder', ->
 				expect(@apiBinder.fetchDevice).to.be.calledTwice
 				@apiBinder.fetchDevice.restore()
 				resinAPI.resinBackend.deviceKeyHandler.restore()
+
+	describe 'offline mode', ->
+		before ->
+			initModels.call(this, '/config-apibinder-offline.json')
+
+		it 'does not generate a key if the device is in offline mode', ->
+			@config.get('offlineMode').then (mode) =>
+				# Ensure offline mode is set
+				expect(mode).to.equal(true)
+				# Check that there is no deviceApiKey
+				@config.get('deviceApiKey').then (key) ->
+					expect(key).to.be.undefined
+
+		describe 'Minimal config offline mode', ->
+			before ->
+				initModels.call(this, '/config-apibinder-offline2.json')
+
+			it 'does not generate a key with the minimal config', ->
```

Comment:
In this case you can also test that a uuid is still generated?

Reply:
I actually meant to add that in and forgot, will add :)